### ISSUE TITLE
licensing

### ISSUE THREAD

**Issue by spotrh** [jtpereyda/libdasm#3]
Hi. There is renewed interest in packaging libdasm for Fedora, however, there are licensing concerns with the codebase.

In essence, the code presents two conflicting license statements:

From README.txt
"libdasm is public domain software. You can do whatever you like with it."

But, the code (and the top of README.txt) says:

"(c) 2004 - 2006 jt / nologin.org"

If something is in the Public Domain, it means that the author has abandoned their copyright on a work. But, if you have a (c) statement, then you are asserting... copyright on the work. You do not have to make that statement to have copyright (thanks to the Berne Convention), but by having it present, it effectively negates the public domain declaration, leaving everyone (except the copyright holder) with no license terms on the work.

There are several ways to fix this. I'll list them in the order of preference:

* Relicense the work under an established open source license. In bz499875, the upstream appeared to be willing to relicense under BSD, which would work fine. This also resolves the international complexity and ambiguity which Public Domain works have. Including a copy of the BSD license text (in README.txt is fine) and noting the BSD license (can be as simple as "license: BSD") in the code header will accomplish this.

* Switch to a Creative Commons Zero license, which is intended to resolve the Public Domain issues by enabling a license when it is not possible to abandon copyright entirely.

* Remove all of the (c) statements from the codebase and README, and replace them with the public domain statement, making it clear that copyright is abandoned on the work.

If you have questions about this, please let me know, I'm happy to advise you on how to resolve the licensing.

**Comment by jtpereyda** [comment]
@spotrh I am not the original author and have only forked this repo. I'm not sure who to track down for an original author.

**Comment by spotrh** [comment]
There is some data regarding that in the Fedora bug ticket: https://bugzilla.redhat.com/show_bug.cgi?id=1559699

**Comment by jtpereyda** [comment]
Given that the original author was clear on the intent to move to the public domain, I'd be happy to update the code to more consistently reflect this fact.

**Comment by xambroz** [comment]
@angea Ange Albertini was probably developper responsible for the https://code.google.com/archive/p/libdasm/

Ange please will you be OK relicensing to BSD type license to get rid some of the uncertainty of the "Public Domain" claim?

-----------------------------------
We have also emails from 2009 from Jarkko Turkulainen <jt@klake.org> and Ero Carrera Ventura <ero.carrera@gmail.com> on the topic of re-licensing to BSD:

-------- Jarkko Turkulainen --------
Subject: 	Re: libdasm/pydasm: licensing question
Date: 	Wed, 15 Jul 2009 07:29:17 +0300
From: 	Jarkko Turkulainen <jt@klake.org>
To: 	David Malcolm <dmalcolm@redhat.com>


Hi!

From my point of view, I see no problem in releasing this with some
BSD-style license. And I'm sure the other guys are with me in this...
The only problem here is that I'm not developing libdasm actively
anymore and I know it's lacking some essential features, like amd64
support.

But, the original "license" still holds, you can include it in Fedora
if you want to, but libdasm support is very limited (= non-existing)
from my side ;-).

-------- Ero Carrera Ventura --------
Subject: 	Re: libdasm/pydasm: licensing question
Date: 	Fri, 17 Jul 2009 00:14:54 -0700
From: 	Ero Carrera Ventura <ero.carrera@gmail.com>
To: 	David Malcolm <dmalcolm@redhat.com>
CC: 	Jarkko Turkulainen <jt@klake.org>


From my part (I was responsible for the Python wrappers) I'm ok with  
any "open" or "free" licensing, I'm not too religious about it and  
definitely want people to make use of it if they find it useful in any  
way.

I'm quite sure that if you ask in the googlecode group the other  
(current) developers will agree.

--
Ero

**Comment by turkja** [comment]
Hello,

I'm the original author. I can now see the funny conflict between "public domain" and (c)... Just to inform, public domain still holds, so I guess removing (c) is the most easy step.

**Comment by xambroz** [comment]
Thank you for your time JT @turkja .

Please could you re-consider changing to some well recognized licensing? The 2-clause BSD or Creative Commons license are usually considered as most permissive (both commercionally and non-commertionally), if that is what you originally intended.

The "public domain" thing is crafted by US. In most of the Europe the copyright law is set in such way that one can't just simply give away rights by not signing the authorship. In such case that code is not authored and explicitly licensed it is assumed that it is protected by the copyright act.
I guess you are from Finland and not from US, if I understand it right - means your code is protected by copyright law by default if you do not grant some rights to use/share/modify etc.

See:
https://fi.wikipedia.org/wiki/Public_domain
https://en.wikipedia.org/wiki/Wikipedia:Public_domain

**Comment by xambroz** [comment]
Please could you consider proposed pull request https://github.com/jtpereyda/libdasm/pull/4 to re-license to BSD 2-clause type license?

**Comment by xambroz** [comment]
I have tried to contact also Ange Albertini (@angea), Ero Carrera Ventura (@erocarrera) and Matt Miller ( skape ... mmiller at hick.org) so they can have a voice about the change.

**Comment by turkja** [comment]
The proposed change in pull request #4 looks perfect for me!

**Comment by angea** [comment]
I'm totally fine with that change.

**Comment by jtpereyda** [comment]
Resolved in PR #4

**Comment by xambroz** [comment]
I got email also from Skape <mmiller@hick.org>:
```
Od: Matt Miller <zAGwf7Wy4VcQ _AT LIVE.COM>
Komu: Michal Ambroz <rebus _AT seznam.cz>, mmiller _AT hick.org <mmiller _AT hick.org>
Datum: 7. 4. 2018 19:10:07
Předmět: RE: libdasm licensing

    Hi Michal,
    For my part I’m OK with this being relicensed as you say below. Feel free to proceed.
    Matt

    From: Michal Ambroz <rebus _AT seznam.cz>
    Sent: Tuesday, March 27, 2018 8:55:31 PM
    To: mmiller _AT hick.org
    Subject: libdasm licensing
     
    Hello Skape,
    I am trying to package libdasm (where you have been contributing to the code) for Fedora Linux distribution.
    Unfortunately I encountered some legal doubts about its "public domain" status during the review of the package - see :
    https://bugzilla.redhat.com/show_bug.cgi?id=499875
    https://bugzilla.redhat.com/show_bug.cgi?id=1559699
    https://github.com/jtpereyda/libdasm/issues/3

    To resolve the doubts I have proposed pull request to the current libdasm repository on github to re-license to 2-clause BSD type license to maintain the "do what you want" status of the package and keep the Fedora lawyers happy.
    In sense of the "public domain" it should be fine, but still it would be really nice if you could explicitly state your approval in the mentioned Issue / pull request on github.
    Please could you consider approving the change on https://github.com/jtpereyda/libdasm/issues/3 ?

    Thank you
    Michal Ambroz
```
